lib/bgs/form686c.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module BGS
+  class Form686c < Service
+    def initialize(user)
+      @user = user.with_indifferent_access
+    end
+
+    def submit(payload)
+      proc_id = create_proc_id_and_form
+      veteran = VnpVeteran.new(proc_id: proc_id, payload: payload, user: @user).create
```

Comment:
I'm noticing a number of service calls, like this one, that chain together multiple external service calls. For example here, `create_proc_id_and_form` itself, calls 2 other external services, which then also might go on and call even more external services. 

I think it'd be clearer if the `service` classes were "dumb" and better adhered to the single-responsibility principle. In this case, submit would require that some kind of a `VnpVeteranCreateResponse` object be passed in.

With dumb services, the chaining logic could be better encapsulated in one place and not scattered through the service classes themselves.

Reply:
I may not be understanding your comment completely so apologies if I'm way off.

The idea I had was to try to bundle some of the calls into a kind of "Bounded Context". Where we have for example, a representation of a veteran in BGS with `VnpVeteran` after a culmination of calls. `Vnp` prefix is probably mis-leading 🤔 

It's my humble opinion that unfortunately, we are shouldering logic we shouldn't have to; many of these calls could/should be lumped together on the service side. So there is indeed heavy chaining going on